app/scripts/lib/network-store.js

```diff
@@ -0,0 +1,42 @@
+const FIXTURE_SERVER_HOST = 'localhost'
+const FIXTURE_SERVER_PORT = 12345
+const FIXTURE_SERVER_URL = `http://${FIXTURE_SERVER_HOST}:${FIXTURE_SERVER_PORT}/state.json`
+
+/**
+ * A read-only network-based storage wrapper
+ */
+module.exports = class ReadOnlyNetworkStore {
+  /**
+   * Returns all of the keys currently saved
+   * @return {Promise<*>}
+   */
+  async get () {
+    try {
+      const response = await fetch(FIXTURE_SERVER_URL)
+      const result = await response.json()
+      return isEmpty(result)
+        ? undefined
+        : result
+    } catch (e) {
+      return undefined
+    }
+  }
+
+  /**
+   * Sets the keys in state
+   * @return {Promise<void>}
+   */
+  async set () {
+    // Pass
+    return Promise.resolve()
```

Comment:
I'm wondering whether ignoring writes will break anything. We could allow writes to take effect in-memory only, without persisting them. I know of at least one place where local storage is relied upon for caching gas estimates, where this fake local storage would result in more network requests than we'd see otherwise.

That's the front-end though. Maybe there are no cases like that in the background.